pkg/network/usm/testutil/libmmap/libmmap.go

```diff
@@ -0,0 +1,35 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package main is a utility to mmap libraries to test shared library handling.
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"golang.org/x/exp/mmap"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Println("usage: libmmap <filename>....")
+		os.Exit(1)
+	}
+
+	for {
+		// To allow time to attach
+		time.Sleep(1000 * time.Millisecond)
```

Comment:
nit: `time.Second`
advanced nit: make that configurable (via command line), the CI can be slow sometimes (and USM won't be fast enough to hook)
advanced advanced nit: The binary can block on getting a character in stdin, to allow the test to signal when to start
advanced nit: how does this file differ from `fmapper` (pkg/network/usm/sharedlibraries/testutil/fmapper/fmapper.go) or `prefetch_file` (pkg/network/usm/testutil/prefetch_file/prefetch_file.go)

on a side note, why do we need both `prefetch_file` and `fmapper`? 😶‍🌫️

Reply:
Thanks, I got rid of the new program and just used `fmapper` which already has a waiting mechanism.